src/PackageManagement.UI/Common/RestartRequestBar.xaml.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Threading;
+using Microsoft.VisualStudio.Shell.Interop;
+using NuGet.Packaging;
+using NuGet.ProjectManagement;
+using VsBrushes = Microsoft.VisualStudio.Shell.VsBrushes;
+
+namespace NuGet.PackageManagement.UI
+{
+    /// <summary>
+    /// Interaction logic for RestartRequestBar.xaml
+    /// </summary>
+    public partial class RestartRequestBar : UserControl, INuGetProjectContext
+    {
+
+        private readonly IDeleteOnRestartManager _deleteOnRestartManager;
```

Comment:
Can you convert this and _vsRestarter to be read-only properties too like the property 'UIDispatcher'?